tensorflow_encrypted/protocol/pond.py

```diff
@@ -895,9 +897,8 @@ def shape(self) -> List[int]:
     def unwrapped(self) -> Tuple[AbstractTensor, ...]:
         return (self.value_on_0, self.value_on_1)
 
-    def eval(self, sess, feed_dict={}, tag=None) -> np.ndarray:
-        value = self.value_on_0.eval(sess, feed_dict=feed_dict, tag=tag)
-        return self.prot._decode(value, self.is_scaled)
+    def decode(self) -> Union[np.ndarray, tf.Tensor]:
```

Comment:
fyi this is a small anti-pattern as we normally have all functionality on the protocol class, and mostly use the tensor objects as structs

Reply:
I see but I can't help to find very strange to write this `t.prot.decode(t)` instead of `t.decode()`